src/kernels/variables/debuggerVariables.ts

```diff
@@ -316,44 +321,14 @@ export class DebuggerVariables
         }
         throw Error('Debugger is not active, cannot evaluate.');
     }
-
-    private async importDataFrameScripts(): Promise<void> {
-        try {
-            // Run our dataframe scripts only once per session because they're slow
-            const key = this.debugService.activeDebugSession?.id;
-            if (key && !this.importedDataFrameScriptsIntoKernel.has(key)) {
-                const scriptPath = DataFrameLoading.getScriptPath(this.context);
-                const contents = await this.fs.readFile(scriptPath);
-                await this.evaluate(contents);
-                this.importedDataFrameScriptsIntoKernel.add(key);
-            }
-        } catch (exc) {
-            traceError('Error attempting to import in debugger', exc);
-        }
-    }
-
-    private async importGetVariableInfoScripts(): Promise<void> {
-        try {
-            // Run our variable info scripts only once per session because they're slow
-            const key = this.debugService.activeDebugSession?.id;
-            if (key && !this.importedGetVariableInfoScriptsIntoKernel.has(key)) {
-                const scriptPath = GetVariableInfo.getScriptPath(this.context);
-                const contents = await this.fs.readFile(scriptPath);
-                await this.evaluate(contents);
-                this.importedGetVariableInfoScriptsIntoKernel.add(key);
-            }
-        } catch (exc) {
-            traceError('Error attempting to import in debugger', exc);
-        }
-    }
-
     public async getFullVariable(variable: IJupyterVariable): Promise<IJupyterVariable> {
-        // See if we imported or not into the kernel our special function
-        await this.importGetVariableInfoScripts();
-
         // Then eval calling the variable info function with our target variable
+        const code = await this.varScriptGenerator.generateCodeToGetVariableInfo({
+            isDebugging: false,
+            variableName: variable.name
+        });
         const results = await this.evaluate(
-            `${GetVariableInfo.VariableInfoFunc}(${variable.name})`,
```

Comment:
The problem here is when debugging we have an await and its possible after we first import the scripts the stack frame has changed, hence the variables will be lost (not in scope).

This fixes that by ensureing that the function call always imports the necessary scripts (previously it was done seprately)

Reply:
THe original code worked because of it loading a module but also had the added benefit of it not needing to be loaded more than once.

Could we change the new code to create a module to accomplish the same thing? So we don't have to load it more than once?

Reply:
Meaning we wouldn't have to recreate the functions on every stack frame.

The side benefit is we would have less extra cruft in the debugger variable list too.

Reply:
I do'nt see any harm in re-creating (re-loading this everytime).
Also we cannot go with a module as that won't work in web/remote.

I'll update the code to delete the imported namespaces & the functions.

Baically change to as follows:
```python
import xyz
def __vsc_getDataFrameInfo():
....

print(results)

delete sys.modules[xyz]
del __vsc_getDataFrameInfo
```

This way we clean up all of the imports and also the function definitions we create.

Reply:
I don't see that as being an issue, its not a large chunk of code that could slow things down, only runs when the user has stopped at a stack frame & running that code is super fast

Reply:
Fixed

Reply:
Oh I guess it is possible. Maybe exec wouldn't work in the debugger though

https://stackoverflow.com/questions/5362771/how-to-load-a-module-from-code-in-a-string

Reply:
This would have the benefit of not really needing the cleanup code too.

Reply:
Yeah but a module wouldn't need to reload on EVERY variable request. It'd be one and done.

Reply:
I think we could leave the module there. We did before.

Reply:
> Could have up to 30 requests for this (assuming variable view was showing 30 variables at a time)

Will check.
FYI - we only have this when debugging, e.g. in regular execution, we send everything as a single request.
Will look into the execution times now.